src/components/shared/chart/Viewport.js

```diff
@@ -3,11 +3,11 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
This file implements a Higher Order Component (HOC), so the typing is quite more difficult than in other places.

What is especially complex is that it both consumes and injects properties.  The documentation at https://flow.org/en/docs/react/hoc/ only shows code that injects properties.

See more context at https://github.com/facebook/flow/issues/4759, there are other solutions but I thought this one was working nicely -- until next version of Flow ? ;)

Reply:
See below for more information too.

Reply:
Note I haven't typed withSize in this patch because it wasn't typed before. But we should.